src/mapfile2js/parse.ts

```diff
@@ -6,84 +6,224 @@ import { checkBlockEndSum } from './parse/checkBlockEndSum';
 import { determineDepth } from './parse/determineDepth';
 import { resolveSymbolset } from './parse/resolveSymbolset';
 
+export interface LineObject {
+  content: string;
+  comment: string;
+  contentWithoutComment: string;
+  key: string;
+  value: string;
+  isBlockKey: boolean;
+  isBlockLine: boolean;
+  depth: number;
+}
+
 /**
- * Parses a MapServer Mapfile to a JavaScript object.
- * @param {string} content Content of a MapServer Mapfile
- * @returns {Object}
+ * A Mapfile can contain a MAP or SYMBOLSET block.
  */
-export function parse(content: string): object {
-  let result = {};
-  const lineObjects: Array<any> = [];
+export interface Mapfile {
+  map: MapfileMap;
+}
 
-  // stack to keep track of blocks
-  const blocks = [result];
 
+interface MapfileSymbolset {
```

Comment:
Can you move all the mapfile element definition in an external `mapfileTypes.ts` file ?

Reply:
Done